R/bind.R

```diff
@@ -0,0 +1,242 @@
+#'
+#' This block allows for row-wise combination of two or more data frames
+#' using [dplyr::bind_rows()]. It stacks data frames vertically, matching
+#' columns by name and filling missing columns with NA values.
+#'
+#' @param add_id Logical, whether to add a column identifying source data frames
+#' @param id_name Character string, name for the ID column (if add_id = TRUE)
+#' @param ... Forwarded to [new_block()]
+#'
+#' @return A block object for bind_rows operations
+#' @export
+new_bind_rows_block <- function(
+  add_id = FALSE,
+  id_name = ".id",
+  ...
+) {
+  new_transform_block(
+    function(id, x, y) {
+      moduleServer(
+        id,
+        function(input, output, session) {
+          # Initialize reactive values
+          r_add_id <- reactiveVal(add_id)
+          r_id_name <- reactiveVal(id_name)
+
+          # Update settings when inputs change
+          observeEvent(input$add_id, {
+            r_add_id(input$add_id)
+          })
+
+          observeEvent(input$id_name, {
+            if (nzchar(input$id_name)) {
+              r_id_name(input$id_name)
+            }
+          })
+
+          # Generate data preview
+          output$data_preview <- renderText({
+            req(x(), y())
+            x_data <- x()
+            y_data <- y()
+
+            x_cols <- colnames(x_data)
+            y_cols <- colnames(y_data)
+            common_cols <- intersect(x_cols, y_cols)
+            x_only <- setdiff(x_cols, y_cols)
+            y_only <- setdiff(y_cols, x_cols)
+
+            preview_text <- paste0(
+              "Left dataset: ", nrow(x_data), " rows, ", ncol(x_data), " columns\n",
+              "Right dataset: ", nrow(y_data), " rows, ", ncol(y_data), " columns\n",
+              "Result: ", nrow(x_data) + nrow(y_data), " rows, ",
+              length(union(x_cols, y_cols)), " columns\n\n"
+            )
+
+            if (length(common_cols) > 0) {
+              preview_text <- paste0(preview_text,
+                "Common columns: ", paste(common_cols, collapse = ", "), "\n")
+            }
+
+            if (length(x_only) > 0) {
+              preview_text <- paste0(preview_text,
+                "Left-only columns: ", paste(x_only, collapse = ", "),
+                " (will be filled with NA in right data)\n")
+            }
+
+            if (length(y_only) > 0) {
+              preview_text <- paste0(preview_text,
+                "Right-only columns: ", paste(y_only, collapse = ", "),
+                " (will be filled with NA in left data)\n")
+            }
+
+            if (r_add_id()) {
+              preview_text <- paste0(preview_text, "\n",
+                "ID column '", r_id_name(), "' will be added to identify source datasets")
+            }
+
+            preview_text
+          })
+
+          # Build bind_rows expression
+          build_bind_expr <- function(add_id, id_name) {
+            if (add_id) {
+              # Create named list with ID labels
+              bquote(
```

Comment:
Here we're using `bquote()` above I saw `parse(text = glue::glue())`. Some consistency would be nice. Not a strict requirement though.

Reply:
now parse(text = glue::glue()) everywhere. Also in blockr.ggplot soon...